File: src/test/java/me/bechberger/ebpf/samples/HelloWorld.java
package me.bechberger.ebpf.samples;

import me.bechberger.ebpf.bcc.BCC;

/**
 * Basic hello world example from hello_world.py
 * <p>
 * {@snippet :
 * from bcc import BPF
 *
 * BPF(text = r"""
 * int kprobe__sys_clone(void *ctx) {
 *     bpf_trace_printk("Hello, World!\\n");
 *     return0;
 * }
 * """).trace_print()
 *}
 */
public class HelloWorld {
    public static void main(String[] args) {
        try (BCC bcc = BCC.builder("""
                int kprobe__sys_clone(void *ctx) {
                   bpf_trace_printk("Hello, World!\\\\n");
                   return0;
                }
                """).build()) {
            bcc.trace_print();
        }
    }
}

File: pysamples/bcc/hello_world.py
#!/usr/bin/python
# source: https://github.com/iovisor/bcc/blob/master/examples/hello_world.py
# formatted and modified by Johannes Bechberger
#
# Copyright (c) PLUMgrid, Inc.
# Licensed under the Apache License, Version 2.0 (the "License")

# run in project examples directory with:
# sudo ./hello_world.py"

from bcc import BPF

BPF(text=r"""
int kprobe__sys_clone(void *ctx) {
    bpf_trace_printk("Hello, World!");
    return 0;
}
""").trace_print()
